src/cli/index.js

```diff
@@ -228,95 +224,141 @@ const getQuestions = ({ appName }) => ({
       },
     },
   ],
-});
+  initial: [
+    {
+      type: 'input',
+      name: 'appPath',
+      message: 'Project directory',
+      askAnswered: true,
+      validate(input) {
+        try {
+          return checkAppPath(input);
+        } catch (err) {
+          console.log();
+          console.error(err.message);
+          return false;
+        }
+      },
+      when(answers) {
+        try {
+          return (
+            answers.interactive &&
+            !answers.config &&
+            !checkAppPath(answers.appPath)
+          );
+        } catch (err) {
+          return true;
+        }
+      },
+    },
+    {
+      type: 'input',
+      name: 'appName',
+      message: 'The name of the application or widget',
+      askAnswered: true,
+      default(answers) {
+        return path.basename(answers.appPath);
+      },
+      validate(input) {
+        try {
+          return checkAppName(input);
+        } catch (err) {
+          console.log();
+          console.error(err.message);
+          return false;
+        }
+      },
+      when(answers) {
+        try {
+          return (
+            answers.interactive &&
+            !answers.config &&
+            !checkAppName(answers.appName)
+          );
+        } catch (err) {
+          return true;
+        }
+      },
+    },
+    {
+      type: 'list',
+      pageSize: 10,
+      name: 'template',
+      message: 'InstantSearch template',
+      askAnswered: true,
+      choices: () => {
+        const templatesByCategory = getTemplatesByCategory();
+
+        return Object.entries(templatesByCategory).reduce(
+          (templates, [category, values]) => [
+            ...templates,
+            new inquirer.Separator(category),
+            ...values,
+          ],
+          []
+        );
+      },
+      validate(input) {
+        // if a config is given, path is optional
+        if (optionsFromArguments.config) {
+          return true;
+        }
 
-async function run() {
-  let appPath = appPathFromArgument;
-  if (!appPath) {
-    const answers = await inquirer.prompt([
-      {
-        type: 'input',
-        name: 'appPath',
-        message: 'Project directory',
+        const isValid = Boolean(input);
+        if (!isValid) {
+          console.log('template is required');
```

Comment:
This `console.log` shouldn't be required, our script errors by itself like with other options, right?

Reply:
it would error in https://github.com/algolia/create-instantsearch-app/blob/c745986d876517dc02bc3ac4374d3c4925bd7ed0/src/cli/getConfiguration.js#L9, but in https://github.com/algolia/create-instantsearch-app/blob/e067902517e97bb85418c7ae9d248b4711cec92a/src/cli/index.js#L339 the validation functions are used before getConfiguration is called, which is why there's two "paths" to requiring template